File: Server/src/main/java/com/augmentedcooking/Models/Rabbitmq/Response/streamed/AiChatStreamedResponse.java
package com.augmentedcooking.Models.Rabbitmq.Response.streamed;

import com.augmentedcooking.Enums.Response.ResponseTypes;
import com.augmentedcooking.Models.Rabbitmq.Response.AiChatResponse;

import lombok.Getter;

@Getter
public abstract class AiChatStreamedResponse<T> extends AiChatResponse {

    private final int sequence;
    private final boolean lastChunk;
    private final T content;

    public AiChatStreamedResponse(String requestId, String type, int sequence, boolean lastChunk, T content) {
        super(requestId, ResponseTypes.valueOf(type));
        this.sequence = sequence;
        this.lastChunk = lastChunk;
        this.content = content;
    }
}

File: Clients/desktop/src/Models/Response/Chat/Streamed/AiChatStreamedResponse.cs
using System.Text.Json.Serialization;
using AugmentedCooking.src.Models.Response.Chat.Serialization;

namespace AugmentedCooking.src.Models.Response.Chat.Streamed;

[JsonConverter(typeof(AiChatStreamedResponseConverter))]
public abstract class AiChatStreamedResponse : AiChatResponse {
    [JsonPropertyName("sequence")]
    public required int Sequence { get; set; }

    [JsonPropertyName("last_chunk")]
    public required bool LastChunk { get; set; }
}

public class AiChatStreamedResponse<T> : AiChatStreamedResponse {
    [JsonPropertyName("content")]
    public required T Content { get; set; }
}
